File: backend/src/domain/services/ShoppingSessionTotalCalculator.ts
export class ShoppingSessionTotalCalculator {
	static calculate(
		items: ReadonlyArray<{ price: number; quantity: number }>,
	): number {
		const rawTotal = items.reduce(
			(sum, item) => sum + item.price * item.quantity,
			0,
		);
		return Math.round(rawTotal * 100) / 100;
	}
}

File: backend-dotnet/src/api/Domain/Services/ShoppingSessionTotalCalculator.cs
namespace SupermarketPlanner.Api.Domain.Services;

public sealed class ShoppingSessionTotalCalculator
{
    public static double Calculate(IReadOnlyList<(double Price, int Quantity)> items)
    {
        var rawTotal = items.Sum(item => item.Price * item.Quantity);
        return Math.Round(rawTotal, 2, MidpointRounding.AwayFromZero);
    }
}
